packages/go_router/lib/src/route.dart

```diff
@@ -448,8 +449,18 @@ class ShellRoute extends RouteBase {
   /// sub-route's builder.
   final ShellRoutePageBuilder? pageBuilder;
 
+  /// The observers for a shell route.
+  ///
+  /// The observers parameter is used by the [Navigator] built for this route.
+  /// sub-route's observers.
+  final List<NavigatorObserver>? observers;
+
   /// The [GlobalKey] to be used by the [Navigator] built for this route.
   /// All ShellRoutes build a Navigator by default. Child GoRoutes
   /// are placed onto this Navigator instead of the root Navigator.
   final GlobalKey<NavigatorState> navigatorKey;
+
+  /// HeroController for each nested Navigator, which solves cases where the
+  /// Hero Widget animation stops working when navigating.
+  final HeroController heroController = HeroController();
```

Comment:
I have asked @angjelkom exactly about this part.
`CupertinoApp` and `MaterialApp` use the static methods `createCupertinoHeroController` and `createMaterialHeroController` accordingly (where the last one sets the specific function to `createRectTween` property of created `HeroController`) to create `HeroController`.
Does `HeroController` creation here need to respect default app behavior? And if yes, it seems that we need to move `HeroController` creation from `ShellRoute` class to `RouteBuilder` class and implement it the same way as a specific page type build for a `builder` function (here, I mean `_pageBuilderForAppType` variable that specifies in `_cacheAppType` depending from `elem` app type).

Reply:
@ycherniavskyi you do have a point there, I will check this after work and let you know, but probably we do need respect default app behaviour and use the platform specific HeroController.